### PR TITLE
Support list file commits

### PR DESCRIPTION
**Pull request by hi-rustin**
We need to get the history of contributors for each file when implementing an external plugin. So add this API to get the commits against the file.

GitHub documentation: https://docs.github.com/en/rest/reference/repos#commits

Example: https://api.github.com/repos/ti-community-infra/tichi/commits?path=README.md

The result is an array of RepositoryCommits.

### CODE REVIEW

File: prow/github/client.go [hunk 1]
```diff
@@ -2969,6 +2970,35 @@ func (c *client) DeleteRef(org, repo, ref string) error {
 	return err
 }
 
+// ListFileCommits returns the commits for this file path.
+//
+// See https://developer.github.com/v3/repos/#list-commits
+func (c *client) ListFileCommits(org, repo, filePath string) ([]RepositoryCommit, error) {
```
Comment (MushuEE) [inline prow/github/client.go:2976]: Would be nice to throw in a quick test for this, the GetRef test is probably a nice framework for it

Comment (hi-rustin) [reply prow/github/client.go:2976]: Added.

**Comment by hi-rustin** [comment]
@alvaroaleman Could you please take a look?

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**